src/ExpressReceiver.ts

```diff
@@ -201,32 +201,72 @@ export default class ExpressReceiver implements Receiver {
     this.bolt = bolt;
   }
 
-  // TODO: the arguments should be defined as the arguments of Server#listen()
-  // TODO: the return value should be defined as a type that both http and https servers inherit from, or a union
-  public start(port: number): Promise<Server> {
+  // TODO: can this method be defined as generic instead of using overloads?
+  public start(port: number): Promise<Server>;
+  public start(portOrListenOptions: number | ListenOptions, serverOptions?: ServerOptions): Promise<Server>;
+  public start(portOrListenOptions: number | ListenOptions, httpsServerOptions?: HTTPSServerOptions): Promise<HTTPSServer>;
+  public start(portOrListenOptions: number | ListenOptions, serverOptions: ServerOptions | HTTPSServerOptions = {}): Promise<Server | HTTPSServer> {
```

Comment:
This is a bit complex ... but probably it should be fine.

Reply:
Any thoughts regarding how it can be simplified?